<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=11"/>
<meta name="generator" content="Doxygen 1.9.3"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>zfcore-demo: zf::ScriptPlayerFunctionJSWrapper Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr id="projectrow">
  <td id="projectalign">
   <div id="projectname">zfcore-demo
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.9.3 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><b>zf</b></li><li class="navelem"><a class="el" href="classzf_1_1_script_player_function_j_s_wrapper.html">ScriptPlayerFunctionJSWrapper</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="classzf_1_1_script_player_function_j_s_wrapper-members.html">List of all members</a>  </div>
  <div class="headertitle"><div class="title">zf::ScriptPlayerFunctionJSWrapper Class Reference</div></div>
</div><!--header-->
<div class="contents">

<p>JS обертка вокруг ScriptPlayerFunction.  
 <a href="classzf_1_1_script_player_function_j_s_wrapper.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="zf__script__player__p_8h_source.html">zf_script_player_p.h</a>&gt;</code></p>

<p>Inherits QObject.</p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a id="pub-methods" name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:ae6fdf4df7561e4cfc3107bf42e643abb"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classzf_1_1_script_player_function_j_s_wrapper.html#ae6fdf4df7561e4cfc3107bf42e643abb">isValid</a> () const</td></tr>
<tr class="memdesc:ae6fdf4df7561e4cfc3107bf42e643abb"><td class="mdescLeft">&#160;</td><td class="mdescRight">Объект инициализирован  <a href="classzf_1_1_script_player_function_j_s_wrapper.html#ae6fdf4df7561e4cfc3107bf42e643abb">More...</a><br /></td></tr>
<tr class="separator:ae6fdf4df7561e4cfc3107bf42e643abb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae84c68786c6600cfb7a3912fb6ef9a1d"><td class="memItemLeft" align="right" valign="top">Q_INVOKABLE QString&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classzf_1_1_script_player_function_j_s_wrapper.html#ae84c68786c6600cfb7a3912fb6ef9a1d">call</a> (QVariantMap args) const</td></tr>
<tr class="memdesc:ae84c68786c6600cfb7a3912fb6ef9a1d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Вызов функции из JS скрипта. Если была ошибка, то всегда возвращает пустую строку. Но обратное не верно, т.е. при возврате пустой строки ошибки может не быть. Надо проверять lastError.  <a href="classzf_1_1_script_player_function_j_s_wrapper.html#ae84c68786c6600cfb7a3912fb6ef9a1d">More...</a><br /></td></tr>
<tr class="separator:ae84c68786c6600cfb7a3912fb6ef9a1d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aba798410438e3f58bbc0b6bd8b1efa66"><td class="memItemLeft" align="right" valign="top">Q_INVOKABLE QString&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classzf_1_1_script_player_function_j_s_wrapper.html#aba798410438e3f58bbc0b6bd8b1efa66">lastError</a> () const</td></tr>
<tr class="memdesc:aba798410438e3f58bbc0b6bd8b1efa66"><td class="mdescLeft">&#160;</td><td class="mdescRight">Ошибка, которая была при вызове call если она есть. Если ошибки нет - пустая строка  <a href="classzf_1_1_script_player_function_j_s_wrapper.html#aba798410438e3f58bbc0b6bd8b1efa66">More...</a><br /></td></tr>
<tr class="separator:aba798410438e3f58bbc0b6bd8b1efa66"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p >JS обертка вокруг ScriptPlayerFunction. </p>
</div><h2 class="groupheader">Member Function Documentation</h2>
<a id="ae84c68786c6600cfb7a3912fb6ef9a1d" name="ae84c68786c6600cfb7a3912fb6ef9a1d"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae84c68786c6600cfb7a3912fb6ef9a1d">&#9670;&nbsp;</a></span>call()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">QString zf::ScriptPlayerFunctionJSWrapper::call </td>
          <td>(</td>
          <td class="paramtype">QVariantMap&#160;</td>
          <td class="paramname"><em>args</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Вызов функции из JS скрипта. Если была ошибка, то всегда возвращает пустую строку. Но обратное не верно, т.е. при возврате пустой строки ошибки может не быть. Надо проверять lastError. </p>

</div>
</div>
<a id="ae6fdf4df7561e4cfc3107bf42e643abb" name="ae6fdf4df7561e4cfc3107bf42e643abb"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae6fdf4df7561e4cfc3107bf42e643abb">&#9670;&nbsp;</a></span>isValid()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">bool zf::ScriptPlayerFunctionJSWrapper::isValid </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Объект инициализирован </p>

</div>
</div>
<a id="aba798410438e3f58bbc0b6bd8b1efa66" name="aba798410438e3f58bbc0b6bd8b1efa66"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aba798410438e3f58bbc0b6bd8b1efa66">&#9670;&nbsp;</a></span>lastError()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">QString zf::ScriptPlayerFunctionJSWrapper::lastError </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Ошибка, которая была при вызове call если она есть. Если ошибки нет - пустая строка </p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li>/home/we/dev/github/zfcore-demo/script_player/private/<a class="el" href="zf__script__player__p_8h_source.html">zf_script_player_p.h</a></li>
<li>/home/we/dev/github/zfcore-demo/script_player/private/zf_script_player_p.cpp</li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by&#160;<a href="https://www.doxygen.org/index.html"><img class="footer" src="doxygen.svg" width="104" height="31" alt="doxygen"/></a> 1.9.3
</small></address>
</body>
</html>
